java-squid/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -57,12 +63,15 @@
    */
   private static final int MAX_STEPS = 2000;
   private static final Logger LOG = LoggerFactory.getLogger(ExplodedGraphWalker.class);
+  private static final Set<String> THIS_SUPER = new HashSet<>(Arrays.asList("this", "super"));
+
   private final ConditionAlwaysTrueOrFalseChecker alwaysTrueOrFalseChecker;
   private ExplodedGraph explodedGraph;
   private Deque<ExplodedGraph.Node> workList;
   private ExplodedGraph.Node node;
   public ExplodedGraph.ProgramPoint programPosition;
   public ProgramState programState;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make programState a static final constant or non-public and provide accessors if needed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)